### ISSUE TITLE
Warn for remote options url in install.rdf

### ISSUE THREAD

**Issue by wagnerand** [mozilla/amo-validator#431]
We should raise a warning for entries in `install.rdf` like:
```
<em:optionsURL>https://somepage.com/settings</em:optionsURL>
```

**Comment by kewisch** [comment]
Maybe it makes sense to file a Firefox bug to just not allow this?

**Comment by wagnerand** [comment]
Feel free to do that :)

**Comment by wagnerand** [comment]
That's what I thought! :)

Adding @eviljeff and @jvillalobos in case someone has strong feelings about this policy change.

**Comment by eviljeff** [comment]
Does it load the options page in a tab that obviously shows the url being remote?

Why would an add-on load a remote page for the options anyway?  It couldn't change anything locally.

**Comment by Noitidart** [comment]
They like to sync things to users account. A very "cloud" approach.

Yep it's clear you're working on a website.

**Comment by eviljeff** [comment]
I think if its clear you're on a website its fine.  Subject to the usual rules around passing data to/from a website that we would apply to the add-on generally.

**Comment by wagnerand** [comment]
So, should this be closed then?

**Comment by Noitidart** [comment]
Playing devils advocate here.

Some possible reasons for going remote, other then total cloud experience: easier on the analytics injections and ad servicing.

It is possible they can change a users pref at any time, but this is true for any online service you have an account with. The large services would have too much fall out (Google, Yahoo, probably Ghostery) so we're safer there. But with smaller guys, with remote prefs, it might be more of a trust issue.

Note on Ghostery: We had them disable the Ghost rank feature in the past, they weren't doing any tricky things to get that Ghost rank behavior back which was good news for trust. Trust is a tricky thing though haha, maybe a policy change to disallow this would probably be a good idea.

**Comment by eviljeff** [comment]
Disallowing remote urls in `<optionsURL>` would achieve nothing other than developers adding their own link/button which loads a website in a tab.  Let's close this.

**Comment by jvillalobos** [comment]
Yeah, I don't see a problem here.

**Comment by erosman** [comment]
There are 2 issues which seems to overlap here:

1. Should remote URLs be allowed in `install.rdf` as `<em:optionsURL>`?
2. More importantly, should addons be allowed to have their preferences at a remote location?

eviljeff's comment covers the first but not the more important second one.
Should the second issue be opened as a separate topic?

**Comment by jvillalobos** [comment]
> More importantly, should addons be allowed to have their preferences at a remote location?

If it's done safely, I don't see it as a problem.

**Comment by erosman** [comment]
IMHO, these are the concerns regarding remote Preferences:

1. User pref can be changed without user consent
2. Risky preferences can be set and since it is remote, it cant be checked during the review
3. Risky preferences can be added and since it is remote, it cant be checked during the review
4. Depending on how often the remote preferences are contacted, it can be used for tracking
5. Depending on the data exchanged with remote preferences on contacts, it can be used for tracking
6. Depending on the code, page content JS has to interact with browser content JS
7. User can be exposed to Ads and/or other undesirable content on addon preference page
8. Aforementioned Ads and/or other undesirable content may pose privacy/security risk (especially in regards to the scope they are executed in)
9. Undesirables content causes user annoyance
10. Additional bandwidth usage just to change preferences
11. Preferences cant be changed while offline
12. Preference cant be changed if the remote server is busy or down

Just a few to note ....
